nixos/modules/system/boot/loader/grub/grub.nix

```
@@ -40,7 +40,7 @@ let
       backgroundColor = f cfg.backgroundColor;
       entryOptions = f cfg.entryOptions;
       subEntryOptions = f cfg.subEntryOptions;
-      grub = f grub;
+      grub = f (if (grub.grubTarget or "") != "" then grub else "");
```

```suggestion
      # on none x86_64-linux like aarch64-linux grubTarget is undefined
      grub = f (if (grub.grubTarget or "") != "" then grub else "");
```
we should comment why we do this especially because you already did a nice investigation

---

Thanks for the suggestion; I added a comment.